src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -581,13 +581,12 @@ validate_args(Args) ->
             ok;
         {[], _, _} ->
             ok;
+        {[Key], StartKey, EndKey} when Key =:= StartKey andalso Key =:= EndKey ->
```

Comment:
If we translate the `key` to `start_key` & `end_key` would we hit this clause? Or would this be for cases (if any) where we may pass `#mrargs` in which have not been validated in parse_param?

Reply:
If this line is removed, two tests in chttpd_db_test will fail.
```
should_return_update_seq_when_set_on_all_docs()
should_not_return_update_seq_when_unset_on_all_docs()
```

Reply:
It's interesting that we didn't need it before. I think it's because we didn't make the key=k keys=[k] and start/end=k exactly equivalent. I made a note of it [here](https://github.com/apache/couchdb/pull/4414/files#r1142357665)

Reply:
Probably, I'll try again.

Reply:
Tried to return Args#mrargs{start_key = Val1, end_key = Val1, `keys=undefined`}, but it would bring another change to `_all_docs`.

If we query with single element keys, and the key is a deleted doc id.
- Before: it will return key-value pair with `deleted:true`.
- After: it will return rows: []

If we don't reset the key to undefined, we just add an exception. Query with keys([Key]), when `Key==start_key && Key==end_key`, return value instead of throwing 400 error.